firehose-writer/src/main/java/com/expedia/www/haystack/pipes/firehoseWriter/AsynchronousFirehoseConsumer.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2018 Expedia, Inc.
+ *
+ *       Licensed under the Apache License, Version 2.0 (the "License");
+ *       you may not use this file except in compliance with the License.
+ *       You may obtain a copy of the License at
+ *
+ *           http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *       Unless required by applicable law or agreed to in writing, software
+ *       distributed under the License is distributed on an "AS IS" BASIS,
+ *       WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *       See the License for the specific language governing permissions and
+ *       limitations under the License.
+ *
+ */
+package com.expedia.www.haystack.pipes.firehoseWriter;
+
+import org.springframework.context.ApplicationContext;
+import org.springframework.core.task.TaskExecutor;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AsynchronousFirehoseConsumer {
+    private final TaskExecutor taskExecutor;
+    private final ApplicationContext applicationContext;
+
+    public AsynchronousFirehoseConsumer(TaskExecutor taskExecutor, ApplicationContext applicationContext) {
+        this.taskExecutor = taskExecutor;
+        this.applicationContext = applicationContext;
+    }
+
+    void executeAsynchronously() {
+        final FirehoseConsumer firehoseConsumer = applicationContext.getBean(FirehoseConsumer.class);
```

Comment:
Minot detail: You can auto wire FireHoseConsumer directly rather than getting form application context.

Reply:
Will change